modules/db/src/main/java/org/mule/module/db/internal/config/domain/query/ParameterizedQueryTemplateFactoryBean.java

```diff
@@ -54,7 +54,11 @@ public QueryTemplate getObject() throws Exception
         {
             QueryParam param = findOverriddenParam(templateParam.getName(), queryParams);
 
-            if (param == null)
+            if (param == null && ((DefaultInputQueryParam) templateParam).isDbInParam())
```

Comment:
Is weird that If something is a DefaultInputQueryParam then you ask for isDbInParam.
I would rename it to hasValue or similar, and must be defined in the InputQueryParam interface. (also note that you can't just cast to DefaultInputQueryParam, you should check that the cast is valid before doing it).